### PR TITLE
Support extension

### PR DESCRIPTION
**Pull request by ConfusedVorlon**
Use Application.shared to allow framework to be linked from extensions.

### CODE REVIEW

File: Sources/Application.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import UIKit
+
+
+internal class Application {
+    static var shared: UIApplication {
```
Comment (OrkhanAlikhanov) [inline Sources/Application.swift:35]: Application should've been `struct` and the file should be in 2 spaced indentation .

**Review by OrkhanAlikhanov — COMMENTED** [review]
Amazing work!

File: Sources/Application.swift (hunk 1 above)
Comment (OrkhanAlikhanov) [reply Sources/Application.swift:35]: And `shared` property should have inline documentation. @danieldahan I think you can merge this in. Just some cosmetic work is left.

**Review by daniel-jonathan — APPROVED** [review]
Thank you! Looks good.

**Pull request merged**